core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -987,13 +991,22 @@ static void loadFileMetadataInternal(RpcContext rpcContext, LockedInodePath inod
       createFileContext.setOperationTimeMs(ufsLastModified);
     }
 
+    boolean useInodeSyncMergeJournalContext
+        = !Configuration.getBoolean(
+            PropertyKey.MASTER_FILE_SYSTEM_MERGE_INODE_JOURNALS
+    );
     try (LockedInodePath writeLockedPath = inodePath.lockFinalEdgeWrite();
-         MergeJournalContext merger = new MergeJournalContext(rpcContext.getJournalContext(),
+         JournalContext merger = useInodeSyncMergeJournalContext
+             ? new MergeJournalContext(rpcContext.getJournalContext(),
              writeLockedPath.getUri(),
-             InodeSyncStream::mergeCreateComplete)) {
+             InodeSyncStream::mergeCreateComplete)
+             : NoopJournalContext.INSTANCE;
+    ) {
       // We do not want to close this wrapRpcContext because it uses elements from another context
-      RpcContext wrapRpcContext = new RpcContext(
-          rpcContext.getBlockDeletionContext(), merger, rpcContext.getOperationContext());
+      RpcContext wrapRpcContext = useInodeSyncMergeJournalContext
+          ? new RpcContext(
+              rpcContext.getBlockDeletionContext(), merger, rpcContext.getOperationContext())
+          : rpcContext;
```

Comment:
do you think you can cleanup these two statements?

Reply:
i think it's okay to replace the existing MergeJournalContext with the new one. This nasty code is used to make sure the existing behavior doesn't change when the feature flag is off. wdty